Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift

```diff
@@ -459,6 +460,88 @@ private extension StripeCardReaderService {
     }
 }
 
+// MARK: - Refunds
+extension StripeCardReaderService {
+    public func refundPayment(parameters: RefundParameters) -> AnyPublisher<String, Error> {
+        if isChipCardInserted {
```

Comment:
I _think_ so, from my testing and experimenting it seemed not-super-happy about issuing refunds with a card already inserted. I will experiment with this more when we have the UI in a better place for trying out the details.

It's a bit more of a balance for refunds than payments: I can see sometimes a refund being done straight away when a transaction was done incorrectly, especially for Simple Payments which could be miskeyed. Perhaps then it's nicer to leave the card in.

However, if a different previous customer has accidentally left their card behind in the reader, it would be quite a thorny problem if that wasn't noticed and the refund was processed on to the previous customer's card, instead of the  right one. It's unlikely, but possible, and if it happened:
* the merchant would be out of pocket (because they'd have to do the refund again for the correct customer) 
* and it would be difficult to do the refund again, because the app/site would think that the order had already been refunded and that can't be undone, as far as I know.

So... I'll check whether it's necessary, and if it's not we can make a choice on what to do.

Reply:
Opened #6461

The more common case will definitely be that the correct customer has put their card in before the merchant hits the refund button, and keeping it as it is right now may be a bit annoying for those customers... but it's only an annoyance so we need to take into account the impact.